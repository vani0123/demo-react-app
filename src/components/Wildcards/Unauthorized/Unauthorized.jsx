import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Paper, Button, Typography } from '@material-ui/core';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { withTranslation } from 'react-i18next';

import { I18N_CONSTANTS } from '../../../constants';

const useStyles = makeStyles((theme) => ({
  container: {
    padding: theme.spacing(2)
  },
  heading: {
    marginBottom: theme.spacing(2)
  },
  content: {
    marginBottom: theme.spacing(4)
  }
}));

const Unauthorized = () => {
  const classes = useStyles();

  return (
    <Paper elevation={2} justify="center" className={classes.container}>
      <div className="no-record">
        <div className={classes.heading}>
          <Typography variant="h5">Oops</Typography>
        </div>
        <div className={classes.content}>
          <Typography variant="body1">
            You are not authorized to view this page.
          </Typography>
        </div>
      </div>
      <Button
        variant="contained"
        color="primary"
        fullWidth
        size="medium"
        disableElevation
      >
        Home
      </Button>
    </Paper>
  );
};

function mapStateToProps(state) {
  return {
    session: state.session
  };
}

Unauthorized.propTypes = {};

const component = connect(mapStateToProps)(Unauthorized);
export default withRouter(withTranslation(I18N_CONSTANTS.NAMESPACE.APP)(component));
