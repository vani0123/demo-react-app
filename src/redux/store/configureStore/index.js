import { createStore, applyMiddleware, compose } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import createSagaMiddleware from 'redux-saga';
import rootReducer from '../../reducers';
import { initSagas } from '../../sagas/initSagas';

export default function configureStore(initialState) {
  const shouldHideDevTool = process.env.NODE_ENV === 'production'
    || typeof document === 'undefined'
    || !window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;

  const composeEnhancers = shouldHideDevTool
    ? compose
    : window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__; // add support for Redux dev tools

  const sagaMiddleware = createSagaMiddleware({
    onError(error) {
      setImmediate(() => {
        throw error;
      });
    }
  });
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancers(
      applyMiddleware(sagaMiddleware, reduxImmutableStateInvariant())
    )
  );

  initSagas(sagaMiddleware);

  return store;
}
