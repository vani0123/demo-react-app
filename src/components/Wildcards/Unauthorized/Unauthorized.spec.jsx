import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from '../../../redux/store/configureStore';
import Unauthorized from './Unauthorized';

const mockFn = jest.fn();

const store = configureStore({
  session: {
    currentTheme: 'DEFAULT',
    hasCheckedSession: true
  },
  common: {
    loadingIndicator: { isLoading: true }
  }
});

const setup = (props) => {
  const defaultProps = {
    t: mockFn,
    correlationId: 'mockedCorrelationId',
    ...props
  };

  return mount(
    <Router>
      <Provider store={store}>
        <Unauthorized {...defaultProps} />
      </Provider>
    </Router>
  );
};

describe('Unauthorized component', () => {
  it('should render correctly', () => {
    const wrapper = setup();

    expect(wrapper.exists()).toBe(true);
  });

  it('should render one Unauthorized tag', () => {
    const wrapper = setup();

    expect(wrapper.find('Unauthorized')).toHaveLength(1);
  });

  it('should have valid t props', () => {
    const wrapper = setup();

    expect(wrapper.find('Unauthorized').props().t).toBe(mockFn);
  });

  it('should have valid snapshot', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
