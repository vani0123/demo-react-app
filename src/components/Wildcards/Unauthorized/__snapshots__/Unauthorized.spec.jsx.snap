// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unauthorized component should have valid snapshot 1`] = `ReactWrapper {}`;
